### PR TITLE
Changes to NuGet.CommandLine to allow transitive restores in NuGet.Packagemanagement.

### PR DESCRIPTION
**Pull request by pranavkm**
cc @emgarten @anurse

[Review on Reviewable](https://reviewable.io/reviews/nuget/nuget3/33)

### CODE REVIEW

File: src/NuGet.CommandLine/Program.cs [hunk 1]
```diff
@@ -74,13 +74,21 @@ public int Main(string[] args)
...
                             var packagesDir = packagesDirectory.HasValue() ?
                                 packagesDirectory.Value() :
-                                Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".dnx", "packages");
+                                Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".nuget", "packages");
                             _log.LogVerbose($"Using packages directory: {packagesDir}");
 
+                            var packageSources = sources.Values.Select(s => new PackageSource(s));
+                            if (!packageSources.Any())
+                            {
+                                var settings = Settings.LoadDefaultSettings(projectDirectory, configFileName: "packages.config", machineWideSettings: null);
```
Comment (analogrelay) [inline src/NuGet.CommandLine/Program.cs]: Is configFileName supposed to be `packages.config` or `nuget.config`?

File: src/NuGet.CommandLine/Program.cs [hunk 2]
```diff
@@ -74,13 +74,21 @@ public int Main(string[] args)
                             // TODO: Do this for real :)
```
Comment (analogrelay) [inline src/NuGet.CommandLine/Program.cs]: With this change, I think we can remove this comment :).

File: src/NuGet.CommandLine/Program.cs (hunk 1 above)
Comment (yishaigalatzer) [inline src/NuGet.CommandLine/Program.cs]: break the long line?

**Comment by yishaigalatzer** [comment]
:shipit: when @anurse is happy

**Comment by emgarten** [comment]
:shipit:

**Comment by pranavkm** [comment]
Updated.

**Comment by pranavkm** [comment]
cc @anurse

File: src/NuGet.CommandLine/Program.cs [hunk 3]
```diff
@@ -49,39 +50,77 @@ public int Main(string[] args)
...
                                 _log.LogVerbose($"Reading project file {projectFile.Value}");
-                                projectDirectory = Path.GetDirectoryName(Path.GetFullPath(projectDirectory));
-                                project = JsonPackageSpecReader.GetPackageSpec(File.ReadAllText(projectFile.Value), Path.GetFileName(projectDirectory), projectFile.Value);
+                                projectPath = Path.GetDirectoryName(projectPath);
+                                project = JsonPackageSpecReader.GetPackageSpec(File.ReadAllText(projectFile.Value), Path.GetFileName(projectPath), projectFile.Value);
+                            }
+                            else if (MsBuildUtility.IsMsBuildBasedProject(projectPath))
+                            {
+#if DNXCORE50
+                                throw new NotSupportedException();
```
Comment (analogrelay) [inline src/NuGet.CommandLine/Program.cs:66]: Seems like you could just have MSBuildUtility throw the NotSupportedException instead and remove one set of #ifdefs

**Comment by analogrelay** [comment]
:rocket: to the :moon: !

:shipit:

**Comment by emgarten** [comment]
:shipit:

File: src/NuGet.Commands/RestoreCommand.cs [hunk 4]
```diff
@@ -51,14 +51,25 @@ public RestoreCommand(ILogger logger)
...
 
             // Load repositories
+            // if (request.Project.FilePath)
+
             var projectResolver = new PackageSpecResolver(request.Project.BaseDirectory);
             var nugetRepository = Repository.Factory.GetCoreV3(request.PackagesDirectory);
 
             var context = new RemoteWalkContext();
 
+            ExternalProjectReference exterenalProjectReference = null;
```
Comment (analogrelay) [inline src/NuGet.Commands/RestoreCommand.cs:61]: nit: typo: 'exterenal'

**Comment by analogrelay** [comment]
One concern that I just thought of. What if I have two projects with the same immediate parent folder name, but in different paths:
- a\c\project.json
- b\c\project.json

This doesn't work by design in DNX land, but in MSBuild land this could totally happen. I wonder if the name of an MSBuild-derived project name should be the full path to the CSProj...

**Comment by yishaigalatzer** [comment]
@pranavkm @anurse do we still need this merged?

**Comment by analogrelay** [comment]
Yes, this or something similar, in order to fix NuGet/Home#610

**Pull request closed without merging**